File: src/main/java/com/truveta/opentoken/tokens/Token.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.tokens;

import java.util.ArrayList;
import com.truveta.opentoken.attributes.AttributeExpression;

/**
 * A token is a collection of attribute expressions that are concatenated
 * together to get the token signature
 * and a unique identifier.
 */
public interface Token {

    /**
     * Get the unique identifier for the token.
     * 
     * @return the unique identifier for the token
     */
    String getIdentifier();

    /**
     * Get the list of attribute expressions that define the token.
     * 
     * @return the list of attribute expressions that define the token
     */
    ArrayList<AttributeExpression> getDefinition();
}

File: lib/nodejs/opentoken/src/tokens/Token.ts
/**
 * Copyright (c) Truveta. All rights reserved.
 */

import { AttributeExpression } from '../attributes/AttributeExpression';

/**
 * A generic interface for tokens.
 *
 * A token is a set of attribute expressions concatenated together using
 * a vertical bar separator (|) to form a token signature.
 *
 * The token signature is then hashed and encrypted to form the final token.
 */
export interface Token {
  /**
   * Gets the unique identifier of the token.
   */
  getIdentifier(): string;

  /**
   * Gets the list of attribute expressions that define the token.
   */
  getDefinition(): AttributeExpression[];
}
